### PR TITLE
Feature/gym 43  create login and singup page

### PR DESCRIPTION
**Pull request by PiotrekBaranowski**
Login / Register page:

- implemented react hook forms and tanstack query
- created UI for page
- Added validation to fields

PC view:
pc-view

PC view (Register):
register

PC view (checked and show password):
checked-and-show-password

Tablet view:
tablet-view

Mobile view:
mobile-view

Created new bugfixes in Jira, because we don't handle BE errors properly

### CODE REVIEW

File: src/app/layout/authenticationPage/loginPage/LoginPage.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    return new Date().getFullYear();
+  }, []);
+
+  const onLogIn = async (data: LoginForm) => {
+    loginUser(data.username, data.password, data.stayLoggedIn)
+      .then(() => {
+        setIsLoggedIn(true);
+        reset();
+      })
+      .catch(() => alert('an error occured'));
```
Comment (Valentine-456) [inline src/app/layout/authenticationPage/loginPage/LoginPage.tsx:34]: Just FYI, I remeber I created for one of my tasks `src\app\components\alerts\AppAlert.tsx` - popup to display messages, warnings, errors, etc

File: src/app/layout/authenticationPage/registerPage/RegisterPage.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    },
+    resolver: yupResolver(registerValidationSchema),
+  });
+
+  const { mutate, error } = useMutation({
+    mutationFn: async (data: Register) => {
+      await registerUser(data);
+    },
+    onSuccess: () => {
+      reset();
```
Comment (Valentine-456) [inline src/app/layout/authenticationPage/registerPage/RegisterPage.tsx:45]: From UX point of view, I would add redirection to login page after success

**Review by Valentine-456 — APPROVED** [review]
It works, it looks good, approve

**Review by przemyslawren — APPROVED** [review]

File: src/app/layout/authenticationPage/loginPage/LoginPage.tsx (hunk 1 above)
Comment (PiotrekBaranowski) [reply src/app/layout/authenticationPage/loginPage/LoginPage.tsx:34]: Done, added previously that task to jira

**Pull request merged**